test/e2e/snaps/test-snap-rpc.spec.js

```diff
@@ -0,0 +1,130 @@
+
+describe('Test Snap RPC', function () {
+  it('can use the cross-snap RPC endowment and produce a public key', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: 25000000000000000000,
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        failOnConsoleError: false,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        // enter pw into extension
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // navigate to test snaps page and connect
+        await driver.driver.get(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+        const snapButton1 = await driver.findElement('#connectRpcSnap');
+        await driver.scrollToElement(snapButton1);
+        await driver.delay(1000);
+        await driver.clickElement('#connectRpcSnap');
+        await driver.delay(1000);
+
+        // switch to metamask extension and click connect
+        let windowHandles = await driver.waitUntilXWindowHandles(
+          2,
+          1000,
+          10000,
+        );
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement(
+          {
+            text: 'Connect',
+            tag: 'button',
+          },
+          10000,
+        );
+
+        await driver.delay(2000);
+
+        // approve install of snap
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Approve & install',
+          tag: 'button',
+        });
+
+        // delay for npm installation
+        await driver.delay(2000);
+
+        // switch back to test snaps page
+        windowHandles = await driver.waitUntilXWindowHandles(1, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // click send inputs on test snap page
+        const snapButton2 = await driver.findElement('#sendRpc');
+        await driver.scrollToElement(snapButton2);
+        await driver.delay(1000);
+        await driver.clickElement('#sendRpc');
+
+        // approve and install part one
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Approve & install',
+          tag: 'button',
+        });
+        await driver.delay(2000);
```

Comment:
same as above

Reply:
I am unsure of how to use the waitForSelector in the case of popups that close and then open another popup opens [this happens in snap installs]
because you have to switch focus back to it before you can search, but if you don’t delay before that search it errors with no such window because of it. 
Should we just leave this delay alone, or is there a similar function to know when a popup window opens?

Reply:
I see. I didn't realise this was a pop-up that closes and automatically opens another one. Maybe here we could use `waitUntilXWindowHandles` to wait for the window to close, then again to wait for the window to reopen. If not then we may need to explore this further later and just go with the delay for now

Reply:
The reason you get the `no such window` error is that the popup is closed, so you would need to switch back to the test dapp if you wanted to wait for something here

Reply:
The extra `waitUntilXWindowHandles` works just fine. Thanks!

Reply:
The added wait worked - thanks!

Reply:
Sounds good to me.

**In the future**, it could be worth experimenting with the `delayStep` of the `waitUntilXWindowHandles` method to see if that helps with the double notification.
An example of this could be reducing the delay from `1000` to `200` e.g. `windowHandles = await driver.waitUntilXWindowHandles(1, 200, 10000);`